sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -202,5 +203,9 @@ private[sql] class JacksonGenerator(
    */
   def write(array: ArrayData): Unit = writeArray(writeArrayData(array, arrElementWriter))
 
+  def write(map: MapData, mapType: MapType): Unit = {
```

Comment:
I am less sure if this `write` should take `mapType`. Looks equivalent `write(row: InternalRow)` does not take the struct type.